homeassistant/components/unifiprotect/binary_sensor.py

```diff
@@ -0,0 +1,263 @@
+            )
+
+    return entities
+
+
+class ProtectDeviceBinarySensor(ProtectDeviceEntity, BinarySensorEntity):
+    """A UniFi Protect Device Binary Sensor."""
+
+    def __init__(
+        self,
+        data: ProtectData,
+        description: ProtectBinaryEntityDescription,
+        device: Camera | Light | Sensor | None = None,
+    ) -> None:
+        """Initialize the Binary Sensor."""
+
+        if device and not hasattr(self, "device"):
+            self.device: Camera | Light | Sensor = device
+        self.entity_description: ProtectBinaryEntityDescription = description
+        super().__init__(data)
+        self._doorbell_callback: CALLBACK_TYPE | None = None
+
+    @callback
+    def _async_update_extra_attrs_from_protect(self) -> dict[str, Any]:
+        attrs: dict[str, Any] = {}
+        key = self.entity_description.key
+
+        if key == _KEY_DARK:
+            return attrs
+
+        if key == _KEY_DOORBELL:
+            assert isinstance(self.device, Camera)
+            attrs[ATTR_LAST_TRIP_TIME] = self.device.last_ring
+        elif isinstance(self.device, Sensor):
+            if key in (_KEY_MOTION, _KEY_DOOR):
+                if key == _KEY_MOTION:
+                    last_trip = self.device.motion_detected_at
+                else:
+                    last_trip = self.device.open_status_changed_at
+
+                attrs[ATTR_LAST_TRIP_TIME] = last_trip
+        elif isinstance(self.device, Light):
+            if key == _KEY_MOTION:
+                attrs[ATTR_LAST_TRIP_TIME] = self.device.last_motion
+
+        return attrs
+
+    @callback
+    def _async_update_device_from_protect(self) -> None:
+        super()._async_update_device_from_protect()
+
+        assert self.entity_description.ufp_value is not None
+
+        self._attr_extra_state_attributes = (
+            self._async_update_extra_attrs_from_protect()
+        )
+
+        if self.entity_description.key == _KEY_DOORBELL:
+            last_ring = get_nested_attr(self.device, self.entity_description.ufp_value)
+            now = utcnow()
+
+            is_ringing = (
+                False if last_ring is None else (now - last_ring) < RING_INTERVAL
+            )
+            _LOGGER.warning("%s, %s, %s", last_ring, now, is_ringing)
+            if is_ringing:
+                if self._doorbell_callback is not None:
+                    _LOGGER.debug("Canceling doorbell ring callback")
+                    self._doorbell_callback()
+                    self._doorbell_callback = None
+                self._doorbell_callback = async_call_later(
+                    self.hass, RING_INTERVAL, self._async_reset_doorbell
+                )
+            self._attr_is_on = is_ringing
+        else:
+            self._attr_is_on = get_nested_attr(
+                self.device, self.entity_description.ufp_value
+            )
+
+    async def _async_reset_doorbell(self, *args: Any, **kwargs: Any) -> None:
```

Comment:
```suggestion
    async def _async_reset_doorbell(self, now: datetime.datetime) -> None:
```
`async_call_later` should always call back with the time.